server/auth/types/kerberos/routes.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { IRouter, SessionStorageFactory, CoreSetup } from 'opensearch-dashboards/server';
+import { sign } from 'jsonwebtoken';
+import { SecuritySessionCookie } from '../../../session/security_cookie';
+import { SecurityPluginConfigType } from '../../..';
+import { User } from '../../user';
+import { SecurityClient } from '../../../backend/opensearch_security_client';
+import { KERBEROS_AUTH_LOGIN, API_AUTH_LOGOUT } from '../../../../common';
+import { resolveTenant } from '../../../multitenancy/tenant_resolver';
+import { AuthType } from '../../../../common';
+
+export class KerberosAuthRoutes {
+  constructor(
+    private readonly router: IRouter,
+    private readonly config: SecurityPluginConfigType,
+    private readonly sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
+    private readonly securityClient: SecurityClient,
+    private readonly coreSetup: CoreSetup
+  ) {}
+
+  public setupRoutes() {
+    // login
+    this.router.get(
+      {
+        path: KERBEROS_AUTH_LOGIN,
+        validate: false,
```

Comment:
shouldn't we validate the path, specifically the nextUrl. see [saml routes setup](https://github.com/DarshitChanpura/security-dashboards-plugin/blob/main/server/auth/types/saml/routes.ts#L54).